learning-bitmark/working-with-bitmarks/using-bitmark-shares.md

```diff
@@ -0,0 +1,631 @@
+      swap -r <receiver> -s <shareId1> -q <quantity1> \
+      -S <shareId2> \
+      -Q <quantity2> \
+      -b <block_number>
+
+
+    # The other owner (the receiver) signs to swap the shares
+    $ bitmark-cli -n <network> -i <receiver identity> \
+      countersign -t <hex-data>
+  ```
+
+  >The `swap` command is to initialize a swap shares request, the request needs to be sign by the sender. Once the receiver signs to accept the request, the swap shares transaction is created and submitted to the blockchain. The `swap` command options:
+  >
+  >* `receiver` - The receiver's identity.
+  >
+  >* `shareId1` - The id of the sender's shares.
+  >
+  >* `quantity1` - The number of shares that the sender wants to swap with the receiver.
+  >
+  >* `shareId2` - The id of the receiver's shares for which the sender wants to swap.
+  >
+  >* `quantity2` - The number of the receiver's shares for which the sender wants to swap.
+  >
+  >* `block_number` - This provides a time-limit to let the request expire and become invalid. Every 30 blocks from the current block adds one hour of expiration.
+  >
+  >
+  >The `countersign` command is used to sign for a data. In this case, it is used by the `receiver` to sign for swapping the shares.
+  >
+  >* `<hex-data>` - The data is returned by the `swap` command.
+  >
+  >
+  >**NOTE:** The grant shares transaction cost a transaction fee of 0.001 LTC or 0.0001 BTC. Only when the fee is paid, the transaction is able to be confirmed on the blockchain.
+
+*Example:*
+
+* One of the owner to initialize the swap shares request
+
+  ```shell
+    $ bitmark-cli -n testing -i first \
+      swap -r second \
+      -s b069f2956b828281dec040782eea3d63793ab4cf17c26f7639e95f6f3b20ba23 \
+      -q 150 \
+      -S 0f366f3ee477a75c6304746f45296d804c08be06ad0941313371a17fe07d778f \
+      -Q 250 \
+      -b 31099
+  ```
+  ```json
+    {
+      "identity": "fUuNhZ6CC4YxUkQB99nuLnUiEevEuwdCoYszJ9Y5uUjp8oiA3A",
+      "swap": "0a20b069f2956b828281dec040782eea3d63793ab4cf17c26f7639e95f6f3b20ba2396012113cbcf0da30e9e1b5acd9cd33f76ef6d93fcec6c81c0a7f3d3fb54d29d3ba8a847200f366f3ee477a75c6304746f45296d804c08be06ad0941313371a17fe07d778ffa012113bf8f8a5e558bdefeaa3fe482dbc9d929e4e6c60ca585e9191e4636bafb9d9144fbf2014078889ed3239b68dc9d78a13c83bd7cd92a8f78e97842d554abb57dea7cc268166228253b094e86273b85010a582e59ee8d5b07d9d9c795c5fe84695b87dcf704"
+    }
+  ```
+
+<br>
+
+* The receiver signs to accept swapping the shares
+
+  ```shell
+    $ bitmark-cli -n testing -i second \
+      countersign -t \
+      0a20b069f2956b828281dec040782eea3d63793ab4cf17c26f7639e95f6f3b20ba2396012113cbcf0da30e9e1b5acd9cd33f76ef6d93fcec6c81c0a7f3d3fb54d29d3ba8a847200f366f3ee477a75c6304746f45296d804c08be06ad0941313371a17fe07d778ffa012113bf8f8a5e558bdefeaa3fe482dbc9d929e4e6c60ca585e9191e4636bafb9d9144fbf2014078889ed3239b68dc9d78a13c83bd7cd92a8f78e97842d554abb57dea7cc268166228253b094e86273b85010a582e59ee8d5b07d9d9c795c5fe84695b87dcf704
+  ```
```

Comment:
If I were trading shares, I'd want to see what the hex said before I signed it. Is there some way to do that? If so, we should list it before we show the signing in the example.

Reply:
Not for now.